src/KeyboardManager.vala

```diff
@@ -17,41 +17,32 @@
 
 namespace Gala {
     public class KeyboardManager : Object {
-        static KeyboardManager? instance;
-        static VariantType sources_variant_type;
+        private static VariantType sources_variant_type;
 
-        public static void init (Meta.Display display) {
-            if (instance != null)
-                return;
-
-            instance = new KeyboardManager ();
-
-            display.modifiers_accelerator_activated.connect (instance.handle_modifiers_accelerator_activated);
-        }
+        public weak Meta.Display display { get; construct; }
+        private GLib.Settings settings;
```

Comment:
If `schema == null` settings does not get explicitly initialized (see line 37). Should be `private Glib.Settings? settings = null;` ?

Reply:
Fields are automatically set to zero at creation time but it might be worth setting it to `GLib.Settings?`